traffic_ops/app/lib/API/Iso.pm

```diff
@@ -203,18 +215,20 @@ sub generate {
 
 	my $iso_url = join( "/", $config->{'to'}{'base_url'}, $iso_dir, $iso_file_name );
 
-	my $response;
-	$response->{isoURL} = $iso_url;
+	my $response = {
+		isoName => $iso_file_name,
+		isoURL => $iso_url,
+	};
 
-	return $self->success( $response, "Generate ISO was successful." );
+	return $response;
 }
 
 sub is_valid {
 	my $self   = shift;
 	my $params = shift;
 
 	my $rules = {
-		fields => [qw/osversionDir hostName domainName rootPass dhcp ipAddress ipNetmask ipGateway ip6Address ip6Gateway interfaceName interfaceMtu disk/],
+		fields => [qw/osversionDir hostName domainName rootPass dhcp ipAddress ipNetmask ipGateway ip6Address ip6Gateway interfaceName interfaceMtu disk mgmtInterface mgmtIpGateway/],
```

Comment:
i think you need to add 2 more things to "fields". probably not a big deal but i think "fields" is supposed to represent ALL params

 mgmtIpAddress 
  mgmtIpNetmask